test/headless/context.js

```diff
@@ -54,4 +54,59 @@ export default class Context {
     this._execution = this._execution.wait(time);
     return this;
   }
+
+  onDOMReady() {
+    this.wait('.tracked-item[data-id="1"]');
+    return this;
+  }
+
+  waitForImpression(identifierNum) {
+    if (identifierNum) {
+      this.wait(`#impression-div-${identifierNum}`);
+    } else {
+      this.wait('#impression-div');
```

Comment:
Can we add a timeout for all of these `wait()` calls? It's important that we ensure there's not a huge delay. I think a timeout of 200ms would be fine.